src/Vegetation/Canopy.jl

```diff
@@ -1,196 +1,643 @@
+model, which is stored in the aux state.
+"""
+function ClimaLSM.surface_temperature(model::CanopyModel, Y, p, t)
+    return model.atmos.T(t)
+end
+
+"""
+    ClimaLSM.surface_height(model::CanopyModel, Y, _...)
+
+A helper function which returns the surface height for the canopy 
+model, which is stored in the parameter struct.
+"""
+function ClimaLSM.surface_height(model::CanopyModel, _...)
+    return model.parameters.h_c
+end
+
+"""
+    ClimaLSM.surface_specific_humidity(model::CanopyModel, Y, p)
+
+A helper function which returns the surface specific humidity for the canopy 
+model, which is stored in the aux state.
+"""
+function ClimaLSM.surface_specific_humidity(
+    model::CanopyModel,
+    Y,
+    p,
+    T_sfc,
+    ρ_sfc,
+)
+    thermo_params =
+        LSMP.thermodynamic_parameters(model.parameters.earth_param_set)
+    return Thermodynamics.q_vap_saturation_generic.(
+        Ref(thermo_params),
+        T_sfc,
+        ρ_sfc,
+        Ref(Thermodynamics.Liquid()),
+    )
+end
+
+"""
+    ClimaLSM.surface_air_density(model::CanopyModel, Y, p)
+    
+A helper function which computes and returns the surface air density for the canopy 
+model.
+"""
+function ClimaLSM.surface_air_density(
+    atmos::PrescribedAtmosphere,
+    model::CanopyModel,
+    Y,
+    p,
+    t,
+    T_sfc,
+)
+    thermo_params =
+        LSMP.thermodynamic_parameters(model.parameters.earth_param_set)
+    ts_in = construct_atmos_ts(atmos, t, thermo_params)
+    return compute_ρ_sfc.(Ref(thermo_params), Ref(ts_in), T_sfc)
+end
 
-A constructor supplying default values for the MedlynConductanceParameters struct.
 """
-function MedlynConductanceParameters{FT}(;
-    Drel = FT(1.6), # unitless
-    g0 = FT(1e-4), # mol/m^2/s 
-    g1 = FT(790), # converted from 5 √kPa to units of √Pa. 5 sqrt(kPa)/(1kPa)*10^3Pa
+    ClimaLSM.surface_evaporative_scaling(model::CanopyModel, Y, p)
+
+A helper function which computes and returns the surface evaporative scaling
+ factor for the canopy model.
+"""
+function ClimaLSM.surface_evaporative_scaling(
+    model::CanopyModel{FT},
+    Y,
+    p,
 ) where {FT}
-    return MedlynConductanceParameters{FT}(Drel, g0, g1)
+    beta = FT(1.0)
+    return beta
+end
+
+
+
+"""
+    DiagnosticTranspiration{FT} <: AbstractTranspiration{FT}
+
+A concrete type used for dispatch when computing the transpiration
+from the leaves, in the case where transpiration is computed
+diagnostically, as a function of prognostic variables and parameters,
+and stored in `p` during the `update_aux!` step.
+"""
+struct DiagnosticTranspiration{FT} <: AbstractTranspiration{FT} end
+
+"""
+    transpiration(transpiration::DiagnosticTranspiration, Y, p, t)
+
+Returns the transpiration computed diagnostically using local conditions.
+In this case, it just returns the value which was computed and stored in 
+the `aux` state during the update_aux! step.
+"""
+function transpiration(transpiration::DiagnosticTranspiration, Y, p, t)
```

Comment:
Including the transpiration functions in the plant hydraulics module would be more intuitive to most land modelers I believe (see offline discussion w Katherine)